### PR TITLE
Use event stream for sick leave email notifications

### PR DESCRIPTION
**Pull request by sponkrashin**
It is for issue #232

### CODE REVIEW

File: server/Arcadia.Assistant.Calendar/SickLeave/SendEmailSickLeaveActor.cs [hunk 1]
```diff
@@ -1,82 +1,55 @@
...
-                            .Ask<EmployeesQuery.Response>(new EmployeesQuery().WithId(notification.CalendarEvent.EmployeeId), TimeSpan.FromSeconds(10))
-                            .ContinueWith(task => new SendNotificationEnd(task.Result, notification.CalendarEvent.Dates))
-                            .PipeTo(this.Self);
+                        this.SendEmail(notification);
                     }
-                    else
+                    catch (Exception ex)
                     {
-                        this.logger.Warning($"Employees actor ref is not set in sending email actor. {Context.Self}");
+                        this.logger.Error(ex.Message);
```
Comment (santee) [inline server/Arcadia.Assistant.Calendar/SickLeave/SendEmailSickLeaveActor.cs]: warning :)

File: server/Arcadia.Assistant.Calendar/SickLeave/EmployeeSickLeaveActor.cs [hunk 2]
```diff
@@ -7,23 +7,25 @@
 
     using Arcadia.Assistant.Calendar.Abstractions;
     using Arcadia.Assistant.Calendar.SickLeave.Events;
+    using Arcadia.Assistant.Notifications.Abstractions;
+    using Arcadia.Assistant.Organization.Abstractions;
 
     public class EmployeeSickLeaveActor : CalendarEventsStorageBase
     {
-        private readonly IActorRef sickLeaveNotifications;
+        private readonly EmployeeMetadata employee;
```
Comment (santee) [inline server/Arcadia.Assistant.Calendar/SickLeave/EmployeeSickLeaveActor.cs]: EmployeeMetadata can change, that's why it stored id only before

Comment (sponkrashin) [reply server/Arcadia.Assistant.Calendar/SickLeave/EmployeeSickLeaveActor.cs]: Changed SickEmail actor notification contract to have only needed fields like employee name and start date. Then sending actor will be responsible to get the data.

File: server/Arcadia.Assistant.Calendar/SickLeave/SendEmailSickLeaveActor.cs (hunk 1 above)
Comment (sponkrashin) [reply server/Arcadia.Assistant.Calendar/SickLeave/SendEmailSickLeaveActor.cs]: Fixed.

File: server/Arcadia.Assistant.Calendar/SickLeave/EmployeeSickLeaveActor.cs (hunk 2 above)
Comment (santee) [reply server/Arcadia.Assistant.Calendar/SickLeave/EmployeeSickLeaveActor.cs]: EmployeeSickLeaveActor might not have proper data, that's what I'm saying. If employee's name changes, or department. You either need to notify this actor about this changes (that's, by the way, possible) or request it somehow.

Comment (sponkrashin) [reply server/Arcadia.Assistant.Calendar/SickLeave/EmployeeSickLeaveActor.cs]: Ok then, let's employee sick actor has reference to it's parent employee actor and gets data before send notification.

**Pull request merged**